src/Docfx.Build/TableOfContents/TocHelper.cs

```diff
@@ -86,7 +86,45 @@ public static TocItemViewModel LoadSingleToc(string file)
             Logger.LogError(message, code: ErrorCodes.Toc.InvalidTocFile);
             throw new DocumentException(message, e);
         }
+    }
+
+    private static bool IsListTocItems(StreamReader reader)
```

Comment:
This looks a bit complicated. For handling YAML, I'm leaning towards we move to a future where YAML are parsed into a JSON node tree, and then deserialized to objects with standard JsonConverter.

Similar to https://github.com/dotnet/docfx/blob/v3/src/docfx/lib/json/YamlUtility.cs#L112 but with System.Text.Json.

Reply:
> This looks a bit complicated.
Remove stream relating unwind logics.
Instead, Read **entire YAML** text first, and then determine TOC type by YAML Parser.